src/flb_http_client.c

```diff
@@ -669,47 +664,79 @@ struct flb_http_client *flb_http_client(struct flb_connection *u_conn,
         break;
     };
 
-    buf = flb_calloc(1, FLB_HTTP_BUF_SIZE);
-    if (!buf) {
-        flb_errno();
-        return NULL;
-    }
-
-    /* FIXME: handler for HTTPS proxy */
     if (proxy) {
         flb_debug("[http_client] using http_proxy %s for header", proxy);
-        ret = snprintf(buf, FLB_HTTP_BUF_SIZE,
-                       fmt_proxy,
-                       str_method,
-                       host,
-                       port,
-                       uri,
-                       flags & FLB_HTTP_10 ? 0 : 1);
+        ret = snprintf(buf, blen,
+                    fmt_proxy,
+                    str_method,
+                    host,
+                    port,
+                    uri,
+                    flags & FLB_HTTP_10 ? 0 : 1);
     }
     else if (method == FLB_HTTP_CONNECT) {
         flb_debug("[http_client] using HTTP CONNECT for proxy: proxy host %s, proxy port %i", host, port);
-        ret = snprintf(buf, FLB_HTTP_BUF_SIZE,
-                       fmt_connect,
-                       str_method,
-                       host,
-                       port,
-                       flags & FLB_HTTP_10 ? 0 : 1);
+        ret = snprintf(buf, blen,
+                    fmt_connect,
+                    str_method,
+                    host,
+                    port,
+                    flags & FLB_HTTP_10 ? 0 : 1);
     }
     else {
         flb_debug("[http_client] not using http_proxy for header");
-        ret = snprintf(buf, FLB_HTTP_BUF_SIZE,
-                       fmt_plain,
-                       str_method,
-                       uri,
-                       flags & FLB_HTTP_10 ? 0 : 1);
+        ret = snprintf(buf, blen,
+                    fmt_plain,
+                    str_method,
+                    uri,
+                    flags & FLB_HTTP_10 ? 0 : 1);
+    }
+    return ret;
+}
+
+struct flb_http_client *flb_http_client(struct flb_connection *u_conn,
+                                        int method, const char *uri,
+                                        const char *body, size_t body_len,
+                                        const char *host, int port,
+                                        const char *proxy, int flags)
+{
+    int ret = 0;
+    char *p;
+    int header_size;
+    int header_align_size;
+    char *buf = NULL;
+    struct flb_http_client *c;
+
+
+    header_size = str_gen_header(method, uri, host, port, proxy, flags, NULL, 0);
```

Comment:
would you please move the composition of this string to use a flb_sds_printf() family functions ?, that will help us to get rid of size calculation

Reply:
If we do that we loose all control over the maximum size of the URL.